### PR TITLE
docs: improve plugins page

### PR DESCRIPTION
**Pull request by ldez**
- fix sidebar items
- add reference file to the doc
- add JSON Schema link for `.custom-gcl.yml`.

### CODE REVIEW

File: scripts/website/expand_templates/linters.go [hunk 1]
```diff
@@ -17,6 +18,20 @@ import (
...
 
+func getExampleSnippets() (*SettingSnippets, error) {
+	reference, err := os.ReadFile(".golangci.reference.yml")
+	if err != nil {
+		return nil, fmt.Errorf("can't read .golangci.reference.yml: %w", err)
+	}
+
+	snippets, err := extractExampleSnippets(reference)
+	if err != nil {
+		return nil, fmt.Errorf("can't read .golangci.reference.yml: %w", err)
```
Comment (alexandear) [inline scripts/website/expand_templates/linters.go:29]:
```suggestion
		return nil, fmt.Errorf("can't extract example snippets from .golangci.reference.yml: %w", err)
```

**Review by alexandear — APPROVED** [review]
Please fix one comment.

**Comment by Gyllsdorff** [comment]
image

@ldez 
I think this might have broken some of the documentation. The [linter](https://golangci-lint.run/usage/linters/)  page worked 30 minutes ago.

**Comment by ldez** [comment]
I will check that.

**Pull request merged**